tracing/src/lib.rs

```diff
@@ -1137,6 +1137,66 @@ pub mod __macro_support {
                 .finish()
         }
     }
+
+    pub mod __tracing_capture_value_by {
+        /// Autoref specialization level 0 for `__tracing_capture_value!`
+        ///
+        /// /!\ WARNING: This is *not* a stable API! /!\
+        /// This type, and all code contained in the `__macro_support` module, is
+        /// a *private* API of `tracing`. It is exposed publicly because it is used
+        /// by the `tracing` macros, but it is not part of the stable versioned API.
+        /// Breaking changes to this module may occur in small-numbered versions
+        /// without warning.
+        #[cfg(all(tracing_unstable, feature = "valuable"))]
+        pub trait TracingCaptureValueByValuable {
+            fn __tracing_capture_value(&self) -> &dyn crate::field::Value;
+        }
+
+        #[cfg(all(tracing_unstable, feature = "valuable"))]
+        impl<T: valuable::Valuable> TracingCaptureValueByValuable for T {
+            fn __tracing_capture_value(&self) -> valuable::Value<'_> {
+                crate::field::valuable(self)
+            }
+        }
+
+        /// Autoref specialization level 1 for `__tracing_capture_value!`
+        ///
+        /// /!\ WARNING: This is *not* a stable API! /!\
+        /// This type, and all code contained in the `__macro_support` module, is
+        /// a *private* API of `tracing`. It is exposed publicly because it is used
+        /// by the `tracing` macros, but it is not part of the stable versioned API.
+        /// Breaking changes to this module may occur in small-numbered versions
+        /// without warning.
+        #[cfg(not(all(tracing_unstable, feature = "valuable")))]
+        pub trait TracingCaptureValueByValue {
+            fn __tracing_capture_value(&self) -> &dyn crate::field::Value;
+        }
+
+        #[cfg(not(all(tracing_unstable, feature = "valuable")))]
+        impl<T: crate::field::Value> TracingCaptureValueByValue for &T {
+            fn __tracing_capture_value(&self) -> &dyn crate::field::Value {
+                self
+            }
+        }
+
+        /// Autoref specialization level 2 for `__tracing_capture_value!`
+        ///
+        /// /!\ WARNING: This is *not* a stable API! /!\
+        /// This type, and all code contained in the `__macro_support` module, is
+        /// a *private* API of `tracing`. It is exposed publicly because it is used
+        /// by the `tracing` macros, but it is not part of the stable versioned API.
+        /// Breaking changes to this module may occur in small-numbered versions
+        /// without warning.
+        pub trait TracingCaptureValueByDebug {
+            fn __tracing_capture_value(&self) -> &dyn crate::field::Value;
+        }
+
+        impl<T: core::fmt::Debug> TracingCaptureValueByDebug for &&T {
+            fn __tracing_capture_value(&self) -> &dyn crate::field::Value {
+                crate::field::debug_ref(self)
```

Comment:
The problem: `&dyn core::fmt::Debug` isn't `&dyn crate::field::Value`. We need to create a `DebugValue<T>` to coerce to `&dyn crate::field::Value` and provide the vtable.

Reply:
I'm not really sure what I can add here, since the problem really is just "`&dyn Debug` is not `&dyn Value`." Change-how-a-value-behind-a-reference-behaves-via-transparent-wrapper is at this point an intrinsic part of my Rust toolkit.